modules/atomic-widgets/props-resolver/props-resolver.php

```diff
@@ -78,13 +78,25 @@ public function resolve( array $schema, array $props ): array {
 			}
 		}
 
-		return array_map(
-			fn( $value ) => $this->transform( $value ),
-			$resolved_props
-		);
+		$transformed_props = [];
+
+		foreach ( $resolved_props as $key => $value ) {
+			$transformed = $this->transform( $value, $key );
+
+			if ( $this->is_multi( $transformed ) ) {
+				unset( $transformed_props[ $key ] );
+
+				$transformed_props = array_merge( $transformed_props, $transformed['value'] );
+				continue;
+			}
+
+			$transformed_props[ $key ] = $transformed;
```

Comment:
You can do something like that:
```php
	private function assign_values( $values ) {
		$resolved = [];

		foreach ( $values as $key => $value ) {
			if ( Multi_Props::is( $value ) ) {
				foreach ( Multi_Props::extract( $value ) as $k => $v ) {
					$resolved[ $k ] = $v;
				}

				continue;
			}

			$resolved[ $key ] = $value;
		}

		return $resolved;
	}
```

then also use it inside the `transform` function

Reply:
I think it makes things complicated.